src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts

```diff
@@ -567,4 +567,19 @@ export class ModelingSubmissionComponent implements OnInit, OnDestroy, Component
 
         return 'entity.action.submitDeadlineMissedTooltip';
     }
+
+    /**
+     * Prepare a result that contains a participation which is needed in the rating component
+     */
+    get resultForRating(): Result | null {
```

Comment:
This is not tied to modeling exercises in specific right? So I am wondering whether it is better to move it in a component that is shared to all other exercise types. This would be useful if you decide to extend the rating feature and allow it for file-upload and programming exercises as well :)

Reply:
For now it is tied to the modeling exercises. I've realized that the text-exercises have a different "result structure" than the modeling exercise. I would suggest to leave it in there and move it if file-upload/programming exercises share the same result structure as the modeling exercises. What do you think about that?

Reply:
Good idea @HanyaElhashemy, this could be in a setter used for the `@Input`.